### PR TITLE
Progress bar

### PR DESCRIPTION
**Pull request by Satttoshi**
please check if my progress bar works

### CODE REVIEW

File: src/zustand/store.js [hunk 1]
```diff
@@ -2,6 +2,16 @@ import { create as createZustand } from "zustand";
...
+  tier: [
+    {
+      id: "tier1",
+      isUnlocked: false,
+      isActive: false,
+      name: "sell Wordpress Website",
+      income: 15,
+      delay: 2000,
+    },
+  ],
```
Comment (djfarly) [inline src/zustand/store.js:14]: Will this `tier` data change over time? It should only be part of state if it is supposed to change

Comment (djfarly) [reply src/zustand/store.js:14]: Otherwise it could just be a `const`

File: src/zustand/store.js [hunk 2]
```diff
@@ -2,6 +2,16 @@ import { create as createZustand } from "zustand";
 
 const useStore = createZustand((set) => ({
```
Comment (djfarly) [inline src/zustand/store.js:3]: I think it is more common to call this `createStore` but it's not super important. :)

File: src/components/MoneyButton.js [hunk 3]
```diff
@@ -1,9 +1,25 @@
-import useStore from "~/src/zustand/store";
+import { useState } from "react";
+
+export default function MoneyButton({
+  addMoney,
+  tier,
+  isFilling,
+  changeFilling,
```
Comment (djfarly) [inline src/components/MoneyButton.js:7]:
```suggestion
  onTimerStart,
  onTimerEnd,
```

File: src/components/Product.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import MoneyButton from "./MoneyButton";
+import ProgressBar from "./ProgressBar";
+import useStore from "~/src/zustand/store";
+import { useState } from "react";
+
+export default function Product() {
+  const { addMoney, tier } = useStore();
+  const [isFilling, changeFilling] = useState(false);
```
Comment (djfarly) [inline src/components/Product.js:8]: a more common naming would be:
```suggestion
  const [isFilling, setIsFilling] = useState(false);
```

File: src/components/Product.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+  function handleChangeFilling() {
+    changeFilling((prev) => !prev);
+  }
```
Comment (djfarly) [inline src/components/Product.js:12]:
```suggestion
  function handleTimerStart() {
    setIsFilling(true);
  }
  
  function handleTimerEnd() {
    setIsFilling(false);
    addMoney();
  }
```

**Review by djfarly — CHANGES_REQUESTED** [review]
Some naming and structure comments below. Otherwise nice progressbar. :)

File: src/components/MoneyButton.js [hunk 6]
```diff
@@ -1,9 +1,25 @@
...
-    <button type="button" onClick={addMoney}>
+    <button
+      type="button"
+      disabled={isFilling}
+      onClick={() => {
+        changeFilling();
+        setTimeout(() => {
+          addMoney();
+          changeFilling();
+        }, delay);
```
[the remark this answers is not in the document]
Comment (Satttoshi) [reply src/components/MoneyButton.js:20]: Thats a very nice hint and looks alot more readable now, thanks!

**Review by FrankSiyli — APPROVED** [review]
LGTM

**Pull request merged**